### PR TITLE
Cavegen: Avoid unsupported biome 'top' or 'filler' nodes 

### PR DESCRIPTION
**Pull request by paramat**
For 3D noise tunnels only, replace unsupported biome 'top' and 'filler'
nodes in tunnel roof with the defined biome 'stone'.
This makes terrain created by 3D noise tunnels consistent with the rest
of mapgen, which does exactly the same in this situation.
////////////

screenshot_20170911_224657

^ Previously the tunnel roof here was unsupported sand

Inspired by #6403 but a simpler implementation (see that issue for details).
Tested.

### CODE REVIEW

File: src/cavegen.cpp [hunk 1]
```diff
@@ -145,6 +156,10 @@ void CavesNoiseIntersection::generateCaves(MMVManip *vm,
 				}
 			} else {
 				// Not tunnel or tunnel entrance floor
+				// Check node for possible replacing with stone for tunnel roof
+				if (c == biome->c_top or c == biome->c_filler)
```
Comment (SmallJoker) [inline src/cavegen.cpp]: `||`

Comment (paramat) [reply src/cavegen.cpp]: Crumbs how embarassing.

**Comment by paramat** [comment]
Fixed silly mistake and retested.

**Comment by paramat** [comment]
7640749d68b87a62868840d34a665101861ca9c0

**Comment by Wuzzy2** [comment]
It's good. :+1:

**Pull request closed without merging**